apps/web/src/components/Button/ButtonWithLink.tsx

```diff
@@ -0,0 +1,58 @@
+"use client"
+
+import Link, { LinkProps } from 'apps/web/node_modules/next/link';
+
+import { Button, ButtonProps, ButtonVariants, ButtonSizes } from './Button';
+
+export function ButtonWithLink({
+  href,
+  target,
+  rel,
+  linkClassNames,
+  linkProps,
+  variant = 'Primary',
+  size = 'Medium',
+  children,
+  buttonClassNames,
+  rounded = false,
+  fullWidth = false,
+  disabled = false,
+  isLoading = false,
+  ...buttonProps
+}: ButtonWithLinkProps) {
+  return (
+    <Link
+      href={href}
+      target={target}
+      rel={rel}
+      className={`${linkClassNames}${disabled ? ' pointer-events-none' : ''}`}
+      {...linkProps}
+    >
+      <Button
+        variant={ButtonVariants[variant]}
+        size={ButtonSizes[size]}
```

Comment:
you should be able to just pass the variant/size directly here.

Reply:
same deal here.

addressed by retyping my implementation to use `ButtonVariants` and `ButtonSizes` types instead of strings.